src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,82 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+
+    public static final String OPERATION_VALUE_DELIMITER = ",";
+    public static final String SUPPLY_OPERATION_TITLE = "supply";
+    public static final String BUY_OPERATION_TITLE = "buy";
+    public static final String RESULT_OPERATION_TITLE = "result";
+    public static final int OPERATION_INDEX = 0;
+    public static final int VALUE_INDEX = 1;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        writeFile(composeReport(calculateOperations(readLinesFromFile(fromFileName))), toFileName);
+    }
+
+    public String[] readLinesFromFile(String filename) {
+        StringBuilder stringBuilder = new StringBuilder();
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(filename))) {
+            String string = bufferedReader.readLine();
+            while (string != null) {
+                stringBuilder.append(string).append(System.lineSeparator());
+                string = bufferedReader.readLine();
+            }
+        } catch (FileNotFoundException e) {
+            throw new RuntimeException("File " + filename + " not found", e);
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read from file " + filename, e);
+        }
+        return String.valueOf(stringBuilder).split(System.lineSeparator());
+    }
+
+    public Table calculateOperations(String[] lines) {
```

Comment:
The method name `calculateOperations` suggests that it calculates multiple operations, but it actually creates a `Table` object with buy and supply totals. Consider renaming it to reflect its purpose more accurately, such as `createTableWithTotals`.